From ced003017fa41220c7973fdd3da8a9e5301a44b3 Mon Sep 17 00:00:00 2001
From: Andreas Geisreiter <ageisreiter@dh-electronics.com>
Date: Wed, 15 Dec 2021 16:03:02 +0100
Subject: [PATCH 69/69] ARM: dts: stm32: Add PicoITX DTO for DH 548-200 DPI
 adapter with MI0700D4T-6 display

Add DT overlay for the PicoITX board for DH 548-200 adapter card with
Multi-Inno MI0700D4T-6 7" DPI display attached to it. This is used to
form a DHMI system.

Upstream-Status: Inappropriate
Signed-off-by: Andreas Geisreiter <ageisreiter@dh-electronics.com>
Signed-off-by: Marek Vasut <marex@denx.de> # deduplication
---
 arch/arm/boot/dts/Makefile                    |  1 +
 ...picoitx-overlay-548-200-x2-mi0700s4t-6.dts | 35 +++++++++++++++++++
 2 files changed, 36 insertions(+)
 create mode 100644 arch/arm/boot/dts/stm32mp15xx-dhcom-picoitx-overlay-548-200-x2-mi0700s4t-6.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index b448afa1f98d..218c1052d3bc 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -1087,6 +1087,7 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp15xx-dhcom-pdk2-overlay-638-100-x12-rpi7inch.dtbo \
 	stm32mp15xx-dhcom-pdk2-overlay-672-100-x18.dtbo \
 	stm32mp157c-dhcom-picoitx.dtb \
+	stm32mp15xx-dhcom-picoitx-overlay-548-200-x2-mi0700s4t-6.dtbo \
 	stm32mp15xx-dhcom-picoitx-overlay-626-100-x2-ch101olhlwh.dtbo \
 	stm32mp157c-dk2.dtb \
 	stm32mp157c-ed1.dtb \
diff --git a/arch/arm/boot/dts/stm32mp15xx-dhcom-picoitx-overlay-548-200-x2-mi0700s4t-6.dts b/arch/arm/boot/dts/stm32mp15xx-dhcom-picoitx-overlay-548-200-x2-mi0700s4t-6.dts
new file mode 100644
index 000000000000..c462c6a08833
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp15xx-dhcom-picoitx-overlay-548-200-x2-mi0700s4t-6.dts
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2021 Andreas Geisreiter <ageisreiter@dh-electronics.com>
+ */
+/dts-v1/;
+/plugin/;
+
+#include "stm32mp15xx-dhcom-overlay-panel-dpi.dtsi"
+
+&display_bl {
+	pwms = <&pwm2 3 3333333 0>; /* 3333333 ns = 300 Hz */
+};
+
+&i2c5 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	eeprom@50 {
+		compatible = "atmel,24c04";
+		reg = <0x50>;
+		pagesize = <16>;
+	};
+};
+
+&ltdc_dpi_out {
+	remote-endpoint = <&panel_in>;
+};
+
+&panel {
+	compatible = "multi-inno,mi0700s4t-6";
+};
+
+&panel_in {
+	remote-endpoint = <&ltdc_dpi_out>;
+};
-- 
2.33.0

